packages/eslint-plugin/src/rules/no-array-delete.ts

```diff
@@ -0,0 +1,57 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import * as ts from 'typescript';
+
+import * as util from '../util';
+
+export default util.createRule<[], 'arrayDeleteViolation'>({
+  name: 'no-array-delete',
+  meta: {
+    docs: {
+      description: 'Disallow delete operator for arrays',
+      recommended: 'error',
+      requiresTypeChecking: true,
+    },
+    messages: {
+      arrayDeleteViolation:
+        'Using delete operator for arrays are forbidden. Use array.splice instead.',
```

Comment:
Nit: just tweaking the phrasing to emphasize that what the user is doing is a bad idea ("dangerous"). In the past we've had user irritation about things described as "forbidden" without any indication that it _should_ be forbidden.

```suggestion
        'Using the delete operator on an array is dangerous. Use array.splice instead.',
```